lib/graphql/execution/execute.rb

```diff
@@ -215,7 +215,12 @@ def resolve_value(value, field_type, field_ctx)
                 result << inner_ctx
                 i += 1
               end
-              result
+
+              if inner_type.non_null? && result.any? { |result| result.value.nil? }
```

Comment:
A more efficient approach could be to short-circuit the `value.each` above if we see any `inner_result.nil?`, but I wasn't sure if it was okay to do this as it might lead to unresolved lazy resolvers. Is that ok?

Reply:
I _think_ it's a _feature_! Right, because instead of doing work, then throwing away the result, we skip the unnecessary work altogether.

Reply:
You're right, short-circuiting is a much better approach as we'll stop resolving child nodes.